### PR TITLE
feat: Setup postgres search

### PR DESCRIPTION
**Pull request by SamoKopecky**
Depends on #133 

- Setup search to use postgres search
- Add configuration option for initial delay and set it to 30 sec for a k8s deployment so that the backend has time to start since the search plugin communicates with the backend to create indexes. [search docs](https://backstage.io/docs/features/search/getting-started#backend)

### CODE REVIEW

**Comment by SamoKopecky** [comment]
Need to test out the config changes in a dev cluster then I will unhold this.

File: packages/backend/src/plugins/search.ts [hunk 1]
```diff
@@ -26,7 +28,7 @@ export default async function createPlugin(
     timeout: { minutes: 15 },
     // A 3 second delay gives the backend server a chance to initialize before
     // any collators are executed, which may attempt requests against the API.
```
Comment (tumido) [inline packages/backend/src/plugins/search.ts:30]: Please remove this comment

**Review by tumido — CHANGES_REQUESTED** [review]

File: packages/backend/src/plugins/search.ts (hunk 1 above)
Comment (SamoKopecky) [reply packages/backend/src/plugins/search.ts]: Sorry, forgot about that. Fixed in the latest commit.

**Review by tumido — APPROVED** [review]
/lgtm
/approve

**Pull request merged**